### PR TITLE
Pipeline rearrange

### PR DESCRIPTION
**Pull request by gtrepta**
#3973 

This factors a few compiler pipeline stages out from between a pair of duplicate stages. If we're able to eliminate the duplicate stages, this change will help narrow down what changes need to be made to do that.

### CODE REVIEW

**Review by Baltoli — APPROVED** [review]

**Comment by ehildenb** [comment]
@gtrepta was this tested downstream with KEVM and KMIR and KWasm? Feel free to add back `automerge` if so!

File: kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java [hunk 1]
```diff
@@ -244,18 +244,18 @@ public Function<Definition, Definition> steps() {
...
             .andThen(resolveFreshConfigConstants)
+            .andThen(generateSortPredicateSyntax)
+            .andThen(generateSortProjections)
+            .andThen(expandMacros)
+            .andThen(AddImplicitComputationCell::transformDefinition)
             .andThen(resolveFreshConstants)
             .andThen(generateSortPredicateSyntax)
             .andThen(generateSortProjections)
+            .andThen(checkSimplificationRules)
             .andThen(subsortKItem)
```
Comment (ehildenb) [inline kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java:259]: I see that this pass is done 2 times as well.

Comment (gtrepta) [reply kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java:259]: `subsortKItem`? Yeah, we identified that in the frontend meeting today, we'll certainly look into it.

**Comment by gtrepta** [comment]
> @gtrepta was this tested downstream with KEVM and KMIR and KWasm? Feel free to add back `automerge` if so!

@ehildenb KEVM and KWasm checked out, their definition.kore files with this change were identical to the ones without the change. KMIR's build system is using some temporary folders which cause different location attributes in the emitted kore, making it harder to determine whether or not they're semantically identical. I'll see if I can isolate the stage of the build that just calls kompile and then make the comparison from there.

**Comment by ehildenb** [comment]
@gtrepta KEVM and KWasm should be good enough I guess, though it would be good to get in the habit of also checking KMIR for stability. I guess adding the relevant files upstreamed here (with `kompile -E` like we've done with KEVM/KWasm) can be done in a follow-up PR to make later changes easier.

**Pull request merged**